### PR TITLE
Verified publishers docs

### PR DESCRIPTION
**Pull request by mit-mit**
Initial publisher docs for pub.dev Verified Publishers.

TODO:
  - [x] Complete steps for creating
  - [x] Link to a sample publisher
  - [x] Add docs for consumers of packages

Staged:
https://kw-www-dartlang-1.firebaseapp.com/tools/pub/publishing
https://kw-www-dartlang-1.firebaseapp.com/tools/pub/verified-publishers

### CODE REVIEW

**Comment by mit-mit** [comment]
fyi @kwalrath @jonasfj started working on this

File: src/tools/pub/publishing.md [hunk 1]
```diff
@@ -52,6 +52,34 @@ Be aware that the email address associated with your Google account is
 displayed on the [Pub site]({{site.pub}}) along with any
 packages you upload.
 
+### Publishing as a pub.dev Verified Publisher
```
[the remark this answers is not in the document]
Comment (mit-mit) [reply src/tools/pub/publishing.md]: @jonasfj I've added two id tags: /publishing#verified-publisher for the overview and 
/publishing#create-verified-publisher for the detailed creation steps

**Comment by mit-mit** [comment]
@kwalrath @jonasfj this is pretty complete now, PTAL

File: src/_data/side-nav.yml [hunk 2]
```diff
@@ -69,6 +69,8 @@
       permalink: /guides/libraries/useful-libraries
     - title: Creating packages
       permalink: /guides/libraries/create-library-packages
+    - title: Verified publishers
```
Comment (kwalrath) [inline src/_data/side-nav.yml]: There's no such page... and no need, right? It seems like we can cover this in the Publishing packages page and maybe the Package reference.

Comment (mit-mit) [reply src/_data/side-nav.yml]: There is now; I forgot to `git add` it yesterday.

This is meant to be the description of what verified publishers are for package *consumers*. The publishing page is for publishers, and has a lot of irrelevant content for consumers.

**Comment by mit-mit** [comment]
Porting comments from https://github.com/dart-lang/site-www/pull/1990#issuecomment-540272208:

@kwalrath wrote:
I did a pretty thorough hatchet job. :) Along the way, I had some questions/thoughts about terminology:

* Can we avoid saying "[verified] publisher **account**"? It sounds like Google Account, but it's just a pub.dev thing, right? Does "account" show up in the pub.dev UI anywhere?
* What's the **user menu**? ("In the user menu in the top-right corner, select **Create Publisher**.") Is there a better name for it? An icon? Alt text? (For formatting the description, I'd follow the precedent in https://developers.google.com/style/ui-elements#term-menus.)
* At one point the text refers to **adding DNS records** ("When adding DNS records, it may take a few hours before the Search Console reflects the changes.") Can we be more general?

**Comment by mit-mit** [comment]
> Can we avoid saying "[verified] publisher account"? 

Yes, we can. But looks like you already made that change?

> What's the user menu? 

It's the menu that opens when you click the photo associated with your user in the top-right corner. We deployed to prod. now, so you should see it if you login on pub.dev?

Really not sure what else we can call this?

> At one point the text refers to adding DNS records ("When adding DNS records, it may take a few hours before the Search Console reflects the changes.") Can we be more general?

Sorry, I don't understand what you mean by general here?

File: src/tools/pub/publishing.md [hunk 3]
```diff
@@ -61,26 +62,68 @@ affect how your package's page looks:
...
+
+1. In the user menu in the top-right corner, select **Create Publisher**.
+
+1. Enter the domain name that you want to associate with your publisher (for example,
+   `dart.dev`), and click **Create Publisher**.
+
+1. In the confirmation dialog, select **OK**.
+
+1. If prompted, complete the verification flow, which opens the [Google
+   Search Console.](https://search.google.com/search-console/about)
```
Comment (jonasfj) [inline src/tools/pub/publishing.md]: Consider: `https://support.google.com/webmasters/answer/34592?hl=en`

File: src/_assets/image/verified-publisher.svg [hunk 4]
```diff
@@ -0,0 +1 @@
+<svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24"><path d="M0 0h24v24H0z" fill="none"/><path d="M12 1L3 5v6c0 5.55 3.84 10.74 9 12 5.16-1.26 9-6.45 9-12V5l-9-4zm-2 16l-4-4 1.41-1.41L10 14.17l6.59-6.59L18 9l-8 8z"/></svg>
```
[the remark this answers is not in the document]
Comment (mit-mit) [reply src/_assets/image/verified-publisher.svg]: Done

**Comment by mit-mit** [comment]
Looks great! Merging...

**Pull request merged**